docs/Open Data Model/Open Data Model.md

```diff
@@ -0,0 +1,755 @@
+|uuids|string|Universally unique identifier(s) comes from thee endpoint context if match|a59ba71e-18b0-f762-2f02-0deaf95xmexzA
+|additional_attrs|string|Additional attributes|Key value pairs
+
+**Network Context Model**
+------------------
+The data model for network context information is based on “whois” information as follows:
+
+|**Attribute**|**Data Type**|**Description**|**Sample Values**|
+|---|---|---|---|
+|domain_name|string|Domain name
+|registry_domain_id|string|Registry Domain ID
+|registrar_whois_server|string|Registrar WHOIS Server
+|registrar_url|string|Registrar URL
+|update_date|bigint|UTC timestamp
+|creation_date|bigint|Creation Date
+|registrar_registration_expiration_date|bigint|Registrar Registration Expiration Date
+|registrar|string|Registrar
+|registrar_iana_id|string|Registrar IANA ID
+|registrar_abuse_contact_email|string|Registrar Abuse Contact Email
+|registrar_abuse_contact_phone|string|Registrar Abuse Contact Phone
+|domain_status|string|Domain Status
+|registry_registrant_id|string|Registry Registrant ID
+|registrant_name|string|Registrant Name
+|registrant_organization|string|Registrant Organization
+|registrant_street|string|Registrant Street
+|registrant_city|string|Registrant City
+|registrant_state_province|string|Registrant State/Province
+|registrant_postal_code|string|Registrant Postal Code
+|registrant_country|string|Registrant Country
+|registrant_phone|string|Registrant Phone
+|registrant_email|string|Registrant Email
+|registry_admin_id|string|Registry Admin ID
+|name_server|string|Name Server
+|dnssec|string|DNSSEC
+
+
+**Extensibility of Data Model**
+==================
+
+The aforementioned data model can be extended to accommodate custom attributes by embedding key-value pairs within the log/alert/context entries. 
+Each model will support an additional attribute by the name of additional_attrs whose value would be a JSON string. This JSON string will contain a Map (and only a Map) of additional attributes that can’t be expressed in the specified model description. Regardless of the type of these additional attributes, they will always be interpreted as String. It’s up to the user, to translate them to appropriate types, if necessary, in the analytics layer. It is also the user’s responsibility to populate the aforementioned attribute as a Map, by presumably parsing out these attributes from the original message.
+For example, if a user wanted to extend the user context model to include a string attribute for “Desk Location” and “City”, the following string would be set for additional_attrs:
+
+|**Attribute Key**|**Attribute Value**|
+|---|---|
+|additional_attrs|{"dsk_location":"B3-F2-W3", "city":"Palo Alto"}|
+
+
+Something similar can be done for endpoint context model, security event log/alert model and other entities.
+
+	Note: This [UDF library](https://github.com/klout/brickhouse) can be used for converting to/from JSON.
```

Comment:
I think the whitespace before Note is going to prevent this from rendering properly.